include/turtle/detail/mutex.hpp

```diff
@@ -65,24 +74,30 @@ namespace mock
 {
 namespace detail
 {
-    struct mutex
+    struct mutex: boost::noncopyable
     {
-        mutex()
-        {}
-        void lock()
-        {}
-        void unlock()
-        {}
+        void lock(){}
+        void unlock(){}
     };
-    struct scoped_lock
+    // Dummy lock classes.
+    // Note: Ctor + Dtor make it RAII classes for compilers and avoid unused variable warnings
+    struct scoped_lock: boost::noncopyable
     {
-        scoped_lock( mutex& )
-        {}
+        scoped_lock( mutex& ){}
+        ~scoped_lock(){}
     };
-    struct lock
+    struct lock: boost::noncopyable
```

Comment:
nit: missing space before `:`

Reply:
This is probably worth having a `class` instead of a `struct` now that it has visibility modifiers?